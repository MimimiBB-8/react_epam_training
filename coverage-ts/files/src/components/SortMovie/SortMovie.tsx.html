
  <!DOCTYPE html>
  <html>
    <head>
      <title>SortMovie.tsx</title>
      <link href="https://cdn.jsdelivr.net/npm/semantic-ui@2.4.2/dist/semantic.min.css" type="text/css" rel="stylesheet">
      <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.58.2/codemirror.min.js" type="text/javascript" charset="utf-8"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.58.2/mode/javascript/javascript.min.js" type="text/javascript" charset="utf-8"></script>
<link href="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.58.2/codemirror.min.css" type="text/css" rel="stylesheet">
<script src="..\..\..\..\assets\source-file.js" type="text/javascript" charset="utf-8"></script>
<link href="..\..\..\..\assets\source-file.css" type="text/css" rel="stylesheet">
    </head>
    <body>
    <div style="margin-top:3em" class="ui container"><h1 class="ui header"><a href="..\..\..\..\index.html">TypeScript coverage report</a></h1><table style="margin-top:2em" class="ui celled table"><thead class=""><tr class=""><th class="">Filename</th><th class="">Percent</th><th class="">Threshold</th><th class="">Total</th><th class="">Covered</th><th class="">Uncovered</th></tr></thead><tbody class=""><tr class="positive"><td class="">src\components\SortMovie\SortMovie.tsx</td><td class="">89.54%</td><td class="">80%</td><td class="">153</td><td class="">137</td><td class="">16</td></tr></tbody></table><textarea id="editor" readonly="" style="margin-top:3em">import { useEffect } from &#x27;react&#x27;;
import style from &#x27;./Sortmovie.module.scss&#x27;
import { useSearchParams } from &#x27;react-router-dom&#x27;

const SortMovie = () =&gt; {

  const [searchParams, setSearchParams] = useSearchParams();

  const elementCarriage = document.querySelector(&#x27;.sort_carriage&#x27;)


  const handleOnClickFilter = (e: React.MouseEvent&lt;HTMLUListElement, MouseEvent&gt;) =&gt; {
    const filter = (e.target as Element).id
    filter === &#x27;all&#x27; ?
      searchParams.delete(&#x27;genre&#x27;) :
      searchParams.set(&#x27;genre&#x27;, filter)
    setSearchParams(searchParams);
    if (elementCarriage) {
      if (elementCarriage.classList.length &gt; 2) {
        elementCarriage.classList.remove(`${elementCarriage.classList[elementCarriage.classList.length - 1]}`)
      }
      elementCarriage.classList.add(`${style[filter]}`)
    }
  }

  const handleOnClickSort = (e: React.ChangeEvent&lt;HTMLSelectElement&gt;) =&gt; {
    const sortParam = e.target.value.split(&#x27;_&#x27;)
    const sortBy = sortParam.slice(0, 2).join(&#x27;_&#x27;)
    const sortOrder = sortParam.slice(-1).join()
    searchParams.set(&#x27;sortBy&#x27;, sortBy);
    searchParams.set(&#x27;sortOrder&#x27;, sortOrder);
    setSearchParams(searchParams);
  }


  return (
    &lt;&gt;
      &lt;div className={style.sort} &gt;
        &lt;nav&gt;
          &lt;ul className={style.sort_genre} onClick={handleOnClickFilter} &gt;
          &lt;li id={&#x27;all&#x27;}&gt;all&lt;/li&gt;
            &lt;li id={&#x27;action&#x27;}&gt;action&lt;/li&gt;
            &lt;li id={&#x27;comedy&#x27;}&gt;comedy&lt;/li&gt;
            &lt;li id={&#x27;horror&#x27;}&gt;horror&lt;/li&gt;
            &lt;li id={&#x27;crime&#x27;}&gt;crime&lt;/li&gt;
          &lt;/ul&gt;
        &lt;/nav&gt;
        &lt;nav&gt;
          &lt;ul className={style.sort_param}&gt;
            &lt;li className={style.sort_by}&gt;SORT BY&lt;/li&gt;
            &lt;li&gt;
              &lt;select className={style.select_date_release} onChange={handleOnClickSort}&gt;
                &lt;option defaultValue=&#x27;release_date&#x27; id={&#x27;release_date&#x27;}&gt;
                  Choose...
                &lt;/option&gt;
                &lt;option value=&#x27;release_date_asc&#x27; id={&#x27;release_date_asc&#x27;}&gt;
                  release ascending
                &lt;/option&gt;
                &lt;option value=&#x27;release_date_des&#x27; id={&#x27;release_date_des&#x27;}&gt;
                  release descending
                &lt;/option&gt;
                &lt;option value=&#x27;vote_average_asc&#x27; id={&#x27;vote_average_asc&#x27;}&gt;
                  rating ascending
                &lt;/option&gt;
                &lt;option value=&#x27;vote_average_desc&#x27; id={&#x27;vote_average_desc&#x27;}&gt;
                  rating descending
                &lt;/option&gt;
              &lt;/select&gt;
            &lt;/li&gt;
          &lt;/ul&gt;
        &lt;/nav&gt;
      &lt;/div&gt;
      &lt;div className={style.sort_identifier}&gt;
        &lt;div className={`sort_carriage ${style.sort_carriage} `}&gt;&lt;/div&gt;
      &lt;/div&gt;
    &lt;/&gt;
  )
}

export default SortMovie
</textarea><pre id="annotations" style="display:none">[{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:1,&quot;character&quot;:7,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:21,&quot;character&quot;:39,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:37,&quot;character&quot;:22,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:37,&quot;character&quot;:28,&quot;text&quot;:&quot;sort&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:39,&quot;character&quot;:25,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:39,&quot;character&quot;:31,&quot;text&quot;:&quot;sort_genre&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:48,&quot;character&quot;:25,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:48,&quot;character&quot;:31,&quot;text&quot;:&quot;sort_param&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:49,&quot;character&quot;:27,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:49,&quot;character&quot;:33,&quot;text&quot;:&quot;sort_by&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:51,&quot;character&quot;:33,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:51,&quot;character&quot;:39,&quot;text&quot;:&quot;select_date_release&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:72,&quot;character&quot;:22,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:72,&quot;character&quot;:28,&quot;text&quot;:&quot;sort_identifier&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:73,&quot;character&quot;:41,&quot;text&quot;:&quot;style&quot;,&quot;kind&quot;:1},{&quot;file&quot;:&quot;src\\components\\SortMovie\\SortMovie.tsx&quot;,&quot;line&quot;:73,&quot;character&quot;:47,&quot;text&quot;:&quot;sort_carriage&quot;,&quot;kind&quot;:1}]</pre></div>
    <p class="footer-text">TypeScript Coverage Report generated by <a href="https://github.com/plantain-00/type-coverage">type-coverage</a> and <a href="https://github.com/alexcanessa/typescript-coverage-report">typescript-coverage-report</a> at Sun, 12 Mar 2023 16:12:31 GMT</p>
    </body>
  </html>
  