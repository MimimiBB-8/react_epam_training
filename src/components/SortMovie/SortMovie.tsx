import style from './Sortmovie.module.scss'
import { useDispatch } from 'react-redux'
import { changeFilterParam, changeSortParam } from '../../store/actions/sort'


const SortMovie = () => {

  const dispatch = useDispatch()

  const handleOnClickFilter = (e: React.MouseEvent<HTMLUListElement, MouseEvent>) => {  
    if((e.target as Element).id !== ''){
      dispatch(changeFilterParam(`&filter=${(e.target as Element).id}`))
    }
  }

  const handleOnClickSort = (e:React.ChangeEvent<HTMLSelectElement>) => {   
    const sortParam = e.target.value.split('_')
    const sortBy = sortParam.slice(0,2).join('_')
    const sortOrder = sortParam.slice(-1).join()
    if(sortParam .length !== 0){
      dispatch(changeSortParam(`&sortBy=${sortBy}&sortOrder=${sortOrder}`))
    }
    
  }

  return (
    <>
      <div className={style.sort}>
        <nav>
          <ul className={style.sort_genre} onClick={handleOnClickFilter}>
            <li id={'all'}>ALL</li>
            <li id={'action'}>Action</li>
            <li id={'comedy'}>COMEDY</li>
            <li id={'horror'}>HORROR</li>
            <li id={'crime'}>CRIME</li>
          </ul>
        </nav>
        <nav>
          <ul className={style.sort_param}>
            <li className={style.sort_by}>SORT BY</li>
            <li>
              <select className={style.select_date_release} onChange={handleOnClickSort}>
                <option defaultValue='release_date' id={'release_date'}>
                  Choose...
                </option>
                <option value='release_date_asc' id={'release_date_asc'}>
                release Ascending
                </option>
                <option value='release_date_des' id={'release_date_des'}>
                release descending
                </option>
                <option value='vote_average_asc' id={'vote_average_asc'}>
                rating Ascending
                </option>
                <option value='vote_average_desc' id={'vote_average_desc'}>
                rating descending
                </option>
              </select>
            </li>
          </ul>
        </nav>
      </div>
      <div className={style.sort_identifier}>
        <div className={style.sort_carriage}></div>
      </div>
    </>
  )
}



export default SortMovie
