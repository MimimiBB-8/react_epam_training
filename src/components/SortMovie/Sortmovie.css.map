{"version":3,"sourceRoot":"","sources":["Sortmovie.sass"],"names":[],"mappings":"AAAA;EACE;EACA;;AAEA;EACE;EACA;EACA;;AACA;EACE;;;AAEN;EACE;EACA;EACA;EACA;EACA;EACA;EACA;;;AAEF;EACE;EACA;EACA","file":"Sortmovie.css"}