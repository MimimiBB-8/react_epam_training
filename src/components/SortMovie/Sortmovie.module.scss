.sort {
  display: flex;
  justify-content: space-between;

  ul {
    display: flex;
    list-style-type: none;
    padding: 0;
    cursor: pointer;
    text-transform: uppercase;
    li:not(:last-child) {
      margin-right: 30px;
    }
  }
}

.sort_identifier {
  width: 100%;
  background-color: #080707;
  border: 1px solid #080707;
  height: 2px;
  margin-bottom: 24px;
  display: flex;
  align-items: center;
}

.sort_carriage {
  width: 35px;
  height: 4px;
  background: #f65261;
}
