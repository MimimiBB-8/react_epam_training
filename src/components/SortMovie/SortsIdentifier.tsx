import React from "react";
import '../../style/SortMovie/sortidettifi.css'
const SortIdentifier: React.FC = () => (
    <div className={''}>

    </div>
);

export default SortIdentifier;
