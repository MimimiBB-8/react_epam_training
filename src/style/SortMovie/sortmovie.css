.sort {
  display: flex;
  justify-content: space-between;
}
.sort ul {
  display: flex;
  list-style-type: none;
  padding: 0;
}
.sort ul li:not(:last-child) {
  margin-right: 30px;
}

/*# sourceMappingURL=sortmovie.css.map */
