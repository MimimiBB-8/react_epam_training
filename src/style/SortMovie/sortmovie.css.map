{"version":3,"sourceRoot":"","sources":["sortmovie.sass"],"names":[],"mappings":"AAAA;EACE;EACA","file":"sortmovie.css"}