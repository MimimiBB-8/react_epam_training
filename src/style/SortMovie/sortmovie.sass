.sort
  display: flex
  justify-content: space-between

  ul
    display: flex
    list-style-type: none
    padding: 0
    li:not(:last-child)
      margin-right: 30px



