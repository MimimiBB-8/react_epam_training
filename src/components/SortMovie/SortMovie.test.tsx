import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SortMovie from "./SortMovie";
import { BrowserRouter as Router } from 'react-router-dom';


describe("When rendering dropdown", () => {
    beforeEach(() => {
        render(
            <Router>
            <SortMovie />
        </Router>);
    });
    it('should display the correct number of options', () => {
        expect(screen.getAllByRole('option').length).toEqual(5)
    })

    it('should correctly set default option', () => {
        expect(screen.getByRole<HTMLOptionElement>('option', { name: 'Choose...' }).selected).toBe(true)
    })
    it('select snapshot', () => {
        const select = render( <Router>
            <SortMovie />
        </Router>)
        expect(select).toMatchSnapshot()
    })
    it('should allow user to change sorting to first option', () => {
        userEvent.selectOptions(
            screen.getByRole('combobox'),
            screen.getByRole('option', { name: 'release ascending' }),
        )
        expect(screen.getByRole<HTMLOptionElement>('option', { name: 'release ascending' }).selected).toBe(true)
    })
    it('should allow user to change sorting to second option', () => {
        userEvent.selectOptions(
            screen.getByRole('combobox'),
            screen.getByRole('option', { name: 'release descending' }),
        )
        expect(screen.getByRole<HTMLOptionElement>('option', { name: 'release descending' }).selected).toBe(true)
    })
    it('should allow user to change sorting to third  option', () => {
        userEvent.selectOptions(
            screen.getByRole('combobox'),
            screen.getByRole('option', { name: 'rating ascending' }),
        )
        expect(screen.getByRole<HTMLOptionElement>('option', { name: 'rating ascending' }).selected).toBe(true)
    })
    it('should allow user to change sorting to fourth option', () => {
        userEvent.selectOptions(
            screen.getByRole('combobox'),
            screen.getByRole('option', { name: 'rating descending' }),
        )
        expect(screen.getByRole<HTMLOptionElement>('option', { name: 'rating descending' }).selected).toBe(true)
    })
});