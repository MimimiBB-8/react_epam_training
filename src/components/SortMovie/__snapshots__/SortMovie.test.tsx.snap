// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When rendering dropdown select snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sort"
      >
        <nav>
          <ul
            class="sort_genre"
          >
            <li
              id="all"
            >
              all
            </li>
            <li
              id="action"
            >
              action
            </li>
            <li
              id="comedy"
            >
              comedy
            </li>
            <li
              id="horror"
            >
              horror
            </li>
            <li
              id="crime"
            >
              crime
            </li>
          </ul>
        </nav>
        <nav>
          <ul
            class="sort_param"
          >
            <li
              class="sort_by"
            >
              SORT BY
            </li>
            <li>
              <select
                class="select_date_release"
              >
                <option
                  id="release_date"
                >
                  Choose...
                </option>
                <option
                  id="release_date_asc"
                  value="release_date_asc"
                >
                  release ascending
                </option>
                <option
                  id="release_date_des"
                  value="release_date_des"
                >
                  release descending
                </option>
                <option
                  id="vote_average_asc"
                  value="vote_average_asc"
                >
                  rating ascending
                </option>
                <option
                  id="vote_average_desc"
                  value="vote_average_desc"
                >
                  rating descending
                </option>
              </select>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="sort_identifier"
      >
        <div
          class="sort_carriage sort_carriage "
        />
      </div>
    </div>
    <div>
      <div
        class="sort"
      >
        <nav>
          <ul
            class="sort_genre"
          >
            <li
              id="all"
            >
              all
            </li>
            <li
              id="action"
            >
              action
            </li>
            <li
              id="comedy"
            >
              comedy
            </li>
            <li
              id="horror"
            >
              horror
            </li>
            <li
              id="crime"
            >
              crime
            </li>
          </ul>
        </nav>
        <nav>
          <ul
            class="sort_param"
          >
            <li
              class="sort_by"
            >
              SORT BY
            </li>
            <li>
              <select
                class="select_date_release"
              >
                <option
                  id="release_date"
                >
                  Choose...
                </option>
                <option
                  id="release_date_asc"
                  value="release_date_asc"
                >
                  release ascending
                </option>
                <option
                  id="release_date_des"
                  value="release_date_des"
                >
                  release descending
                </option>
                <option
                  id="vote_average_asc"
                  value="vote_average_asc"
                >
                  rating ascending
                </option>
                <option
                  id="vote_average_desc"
                  value="vote_average_desc"
                >
                  rating descending
                </option>
              </select>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="sort_identifier"
      >
        <div
          class="sort_carriage sort_carriage "
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sort"
    >
      <nav>
        <ul
          class="sort_genre"
        >
          <li
            id="all"
          >
            all
          </li>
          <li
            id="action"
          >
            action
          </li>
          <li
            id="comedy"
          >
            comedy
          </li>
          <li
            id="horror"
          >
            horror
          </li>
          <li
            id="crime"
          >
            crime
          </li>
        </ul>
      </nav>
      <nav>
        <ul
          class="sort_param"
        >
          <li
            class="sort_by"
          >
            SORT BY
          </li>
          <li>
            <select
              class="select_date_release"
            >
              <option
                id="release_date"
              >
                Choose...
              </option>
              <option
                id="release_date_asc"
                value="release_date_asc"
              >
                release ascending
              </option>
              <option
                id="release_date_des"
                value="release_date_des"
              >
                release descending
              </option>
              <option
                id="vote_average_asc"
                value="vote_average_asc"
              >
                rating ascending
              </option>
              <option
                id="vote_average_desc"
                value="vote_average_desc"
              >
                rating descending
              </option>
            </select>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="sort_identifier"
    >
      <div
        class="sort_carriage sort_carriage "
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
